FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheck/HealthCheckBackgroundService.cs

```diff
@@ -71,15 +71,17 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
                 catch (OperationCanceledException e)
                 {
                     // not expected to trigger this exception.
-                    _diagnosticLogger.LogError($"Health check service is cancelled. {e.Message}");
-                    _logger.LogInformation(e, $"Health check service is cancelled. {e.Message}");
+                    _diagnosticLogger.LogError($"Health check service is cancelled. Reason: {e.Message}");
+                    _logger.LogInformation(e, $"Health check service is cancelled. Reason: {e.Message}");
+                    _metricsLogger.LogTotalErrorsMetrics(ErrorType.HealthCheckError, $"Health check service is cancelled. Reason: {e.Message}", Operations.HealthCheck);
                     await delayTask;
                     throw;
```

Comment:
who catches the cancellation exception and handles it?

Reply:
Will not handle and health checker service will stop.